ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/reader/parquet/ParquetStreamReader.scala

```diff
@@ -76,16 +73,25 @@ private[reader] class ParquetStreamReader(
 
 object ParquetStreamReader extends StreamReaderFactory with ParquetStreamReaderAttributes {
   private val logger = LogManager.getLogger
-  private val defaultCheckForInitialFileInterval = "10000"
+  private val DefaultCheckForInitialFileInterval = "10000"
+  private val DefaultWaitForFiles = false
 
   override def apply(conf: Configuration): StreamReader = {
     ComponentFactoryUtil.validateConfiguration(conf, getProperties)
 
     val path = conf.getString(KeySourceDirectory)
-    val waitForFiles = Try(conf.getString(KeyWaitForFiles)).toOption.isDefined
-    val checkForInitialFileInterval = Try(conf.getString(KeyCheckForInitialFileInterval))
-      .getOrElse(defaultCheckForInitialFileInterval)
-      .toLong
+    val waitForFiles = ConfigUtils.getBooleanOrNone(KeyWaitForFiles, conf) match {
+      case Success(value) => value.getOrElse(DefaultWaitForFiles)
+      case Failure(exception) => throw exception
+    }
+    val checkForInitialFileInterval = Try(ConfigUtils.getOrNone(KeyCheckForInitialFileInterval, conf)) match {
+      case Success(value) => value.getOrElse(DefaultCheckForInitialFileInterval).toLong
+      case Failure(exception) => throw exception
+    }
```

Comment:
I may have missed something, but it seems to me like this could be solved by: 
```
conf.getLong(KeyCheckForInitialFileInterval, DefaultCheckForInitialFileInterval)
```

While changing DefaultCheckForInitialFileInterval type to Long